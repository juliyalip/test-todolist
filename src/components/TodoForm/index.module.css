.formContainer {
  background-color: var(--primary-background-color);
  border: 1px solid var(--input-border-color);
  border-radius: 4px;
  width: 100%;
  padding-left: 12px;
  margin-bottom: 10px;
  box-shadow: var(--heavy-shadow);
  transition: border-color 250ms ease;
}
@media screen and (min-width: 768px) {
  .formContainer {
    margin-bottom: 15px;
  }
}
@media screen and (min-width: 1200px) {
  .formContainer {
    margin-bottom: 20px;
  }
}

.activeFormContainer {
  border-color: var(--input-focus-color);
}

.inputElement {
  width: 80%;
  padding: 10px 5px;
  font-size: 0.9rem;
  vertical-align: middle;
  outline: 0;
  border: 0;
}
@media screen and (min-width: 768px) {
  .inputElement {
    padding: 15px 15px;
  }
}
@media screen and (min-width: 1200px) {
  .inputElement {
    padding-top: 17px;
    padding-bottom: 17px;
  }
}

.labelForActiveForm {
  display: inline-block;
  width: 17px;
  height: 17px;
  border-radius: 50%;
  background-color: white;
  border: 1px solid var(--input-border-color);
  vertical-align: middle;
  margin-right: 7px;
  opacity: 0;
  transition: opacity 250ms ease;
}
@media screen and (min-width: 1200px) {
  .labelForActiveForm {
    width: 20px;
    height: 20px;
    margin-right: 5px;
  }
}/*# sourceMappingURL=index.module.css.map */