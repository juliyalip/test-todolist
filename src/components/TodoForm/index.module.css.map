{"version":3,"sources":["index.module.scss","index.module.css"],"names":[],"mappings":"AAAA;EACE,iDAAA;EACA,2CAAA;EACA,kBAAA;EACA,WAAA;EACA,kBAAA;EACA,mBAAA;EACA,+BAAA;EACA,mCAAA;ACCF;ADEA;EAXA;IAYE,mBAAA;ECCA;AACF;ADCE;EAfF;IAgBI,mBAAA;ECEF;AACF;;ADCA;EACE,sCAAA;ACEF;;ADCA;EACE,UAAA;EACA,iBAAA;EACA,iBAAA;EACA,sBAAA;EACA,UAAA;EACA,SAAA;ACEF;ADAE;EARF;IASI,kBAAA;ECGF;AACF;ADFG;EAXH;IAYI,iBAAA;IACA,oBAAA;ECKF;AACF;;ADDA;EACE,qBAAA;EACA,WAAA;EACA,YAAA;EACA,kBAAA;EACA,uBAAA;EACA,2CAAA;EACA,sBAAA;EACA,iBAAA;EACA,UAAA;EACA,8BAAA;ACIF;ADHE;EAXF;IAYI,WAAA;IACA,YAAA;IACA,iBAAA;ECMF;AACF","file":"index.module.css"}