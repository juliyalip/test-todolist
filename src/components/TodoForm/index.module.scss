.container {
    width: 350px;
    margin-left: auto;
    margin-right: auto;
    padding: 10px 15px;
    border: 1px solid black;
    border-radius: 4px;
    display: flex;
    flex-direction: column;
    box-shadow: 7px 7px grey,  ;
}

.elForm{
    margin-bottom: 15px;
}