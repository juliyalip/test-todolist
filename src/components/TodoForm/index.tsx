import React, { useState , useEffect} from 'react';
import { useAppDispatch } from '../../redux/hooks';
import shortid from 'shortid';
import { addTodo } from '../../redux/todos-slice';
import Alert from '../../components/itoms/Alert';

import style from './index.module.scss';

const TodoForm = () => {
  const dispatch = useAppDispatch();
  const [text, setText] = useState<string>('');
  const [visibleAlert, setVisibleAlert] = useState(false);
  const changeText = (e: React.ChangeEvent<HTMLInputElement>) => {
    setText(e.target.value);
  };

useEffect(()=>{
setTimeout(()=>{
  setVisibleAlert(false)
}, 2000)
}, [visibleAlert])

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (!text) {
      setVisibleAlert(true);
      return;
    }
    dispatch(addTodo({ text, id: shortid.generate(), done: false }));

    setText('');
  };

  return (
    <>
      {visibleAlert && <Alert />}
      <form onSubmit={handleSubmit} className={style.container}>
        {text && <span className={style.labelForActiveForm}></span>}

        <input
          type="text"
          value={text}
          onChange={changeText}
          className={style.inputElement}
        />
      </form>
    </>
  );
};

export default TodoForm;
