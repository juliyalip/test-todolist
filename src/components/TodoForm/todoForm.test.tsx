import { fireEvent } from '@testing-library/react';
import { renderWithProviders } from 'utils/test-utils';
import TodoForm from './index';

const renderComponent = () => renderWithProviders(<TodoForm />);

describe('test todoForm', () => {
  it('test form element render', () => {
    const { getByTestId } = renderComponent();
    const input = getByTestId('input');

    expect(input.textContent).toBe('');
    expect(input).toBeValid();
  });

  it('update input element', () => {
    const input = document.querySelector('input');
    if (input) {
      fireEvent.input(input, { target: { value: 'hello' } });

      expect(input?.value).toBe('hello');
    }
  });

  it('conditional for render  button element', () => {
    const { getByTestId } = renderComponent();
    const input = getByTestId('input');
    if (input) {
      fireEvent.input(input, { target: { value: 'hello' } });
    }
    const buttonElement = getByTestId('button');

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toMatchSnapshot();
  });
});
