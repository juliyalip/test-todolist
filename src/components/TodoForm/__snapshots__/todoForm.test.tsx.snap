// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test todoForm conditional render for button element 1`] = `
<button
  class="btnOnForm"
  type="submit"
/>
`;
